domains/certificates/Query.API/API/ContractService/Repositories/WalletRepository.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using DataContext;
+using DataContext.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.ContractService.Repositories;
+
+public interface IWalletRepository
+{
+    Task AddWallet(Wallet wallet);
+    Task<Wallet?> GetWalletByOwnerSubject(Guid ownerSubject, CancellationToken cancellationToken);
+}
+
+public class WalletRepository : IWalletRepository
+{
+    private readonly ApplicationDbContext dbContext;
+
+    public WalletRepository(ApplicationDbContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+
+    public async Task AddWallet(Wallet wallet)
+    {
+        await dbContext.Wallets.AddAsync(wallet);
+    }
+
+    public async Task<Wallet?> GetWalletByOwnerSubject(Guid ownerSubject, CancellationToken cancellationToken)
```

Comment:
Are there any conventions here, like get* method will throw if not existing, otherwise use a find* method?

Reply:
This has been deleted in v4 - we get the wallet from WalletSystem. Not sure I understand the question tho :)